### DISCUSSION TITLE
Invalid Language Id; Table: File, Column: Language, Key(s)

### DISCUSSION THREAD

**Discussion by kcs-omkarsutar** [wixtoolset/issues#8644] [WiX Development]
hii,

I am trying to make an msi installer for a maui windows application.
I get some warning messages like:

WIX1076 WARNING: ICE03: string overflow (exceeds length allowed in column); Table: File, Column: Language, Key(s): fil5F8446D4678F6AF100904EC0512845FE

WIX1076 WARNING: ICE60: file fil2D5C96A25B27C35446AAF23DD4239481 is not a font, and its version is not an associate file reference. It should contain the language specified in the Language column.

I tried to find a solution but I didn't succeed.
If you have any suggestions on this, it will help me a lot in future development.

**Comment by bevanweiss** [comment]
You probably need to look to Orca here.
For the [ICE03](https://learn.microsoft.com/en-us/windows/win32/msi/ice03), it's saying that in the File table, you've authored a row which has a value in the Language column which is too long.  And it's in the row with File Id = fil5F8446D4678F6AF100904EC0512845FE
https://learn.microsoft.com/en-us/windows/win32/msi/file-table

For the [ICE60](https://learn.microsoft.com/en-us/windows/win32/msi/ice60).  You have a file (referenced by row fil2D5C96A25B27C35446AAF23DD4239481), it has a version associated with it (so it will be a  versioned file), but it does not have a language defined in the file table which matches the language within the file.
The warning is telling you that if the user does any installer operation that assesses the state of this file, it will be marked as not matching the installer, since the file language + version combo on disk will not match the installer record of the language + version combo.
You could configure the language, or you could remove the version.

When submitting items that involve your code, you should include your code... it stops us from having to guess what you've done to cause the error / warning.

I suspect in this situation you've used the fileID reference for both the Version and Language fields.
Whilst that's ok for the Version column, it's not ok for the Language column.
You should consult the [File table](https://learn.microsoft.com/en-us/windows/win32/msi/file-table) documentation to see what kind of values it wants for different behaviours.
